### ISSUE TITLE
Package under check not available via Rscript for Windows

### ISSUE THREAD

**Issue by lorenzwalthert** [r-lib/rcmdcheck#136]
In `{precommit}` tests, I run `Rscript` via [`system2()`](https://github.com/lorenzwalthert/precommit/blob/master/R/testing.R#L116-L119) and these R files contain calls to the `{precommit}` package, i.e. the package I am checking. At least that's what I wanted to introduce in https://github.com/lorenzwalthert/precommit/pull/225. Everything worked fine before I called `{precommit}` in these scripts.

However, now I get `Error: Error in loadNamespace(name) : there is no package called 'precommit'` [on Windows](https://github.com/lorenzwalthert/precommit/runs/1939109479#step:13:230) only with [standard workflows](https://github.com/lorenzwalthert/precommit/blob/master/.github/workflows/R-CMD-check.yaml) containing
```r
rcmdcheck::rcmdcheck(args = c("--no-manual", "--as-cran"), error_on = "warning", check_dir = "check")
```
macOS and Ubuntu are fine as you can see from the above link.

I was not sure the problem was a limitation of `R CMD Check` or a problem of `{rcmdcheck}`. I ran `devools::check_win_release()` and the [log](https://win-builder.r-project.org/X7nHvrVUqVxV/00check.log) does not show these errors (but some other unrelated ones) (see output below), so I assume it's a bug of `{rcmdcheck}`. Maybe something with library paths? Maybe a reprex would be helpful. It's a bit of work, so I thought I'd first post this with relevant links and wait for your feedback. 

My workaround is run `remotes::install_local()` before the check.

***

```
* using log directory 'd:/RCompile/CRANguest/R-release/precommit.Rcheck'
* using R version 4.0.4 (2021-02-15)
* using platform: x86_64-w64-mingw32 (64-bit)
* using session charset: ISO8859-1
* checking for file 'precommit/DESCRIPTION' ... OK
* this is package 'precommit' version '0.1.3.9000'
* package encoding: UTF-8
* checking CRAN incoming feasibility ... NOTE
Maintainer: 'Lorenz Walthert <lorenz.walthert@icloud.com>'

Version contains large components (0.1.3.9000)

Unknown, possibly mis-spelled, fields in DESCRIPTION:
  'Remotes'
* checking package namespace information ... OK
* checking package dependencies ... OK
* checking if this is a source package ... OK
* checking if there is a namespace ... OK
* checking for hidden files and directories ... OK
* checking for portable file names ... OK
* checking serialization versions ... OK
* checking whether package 'precommit' can be installed ... OK
* checking installed package size ... OK
* checking package directory ... OK
* checking for future file timestamps ... OK
* checking 'build' directory ... OK
* checking DESCRIPTION meta-information ... OK
* checking top-level files ... OK
* checking for left-over files ... OK
* checking index information ... OK
* checking package subdirectories ... OK
* checking R files for non-ASCII characters ... OK
* checking R files for syntax errors ... OK
* loading checks for arch 'i386'
** checking whether the package can be loaded ... OK
** checking whether the package can be loaded with stated dependencies ... OK
** checking whether the package can be unloaded cleanly ... OK
** checking whether the namespace can be loaded with stated dependencies ... OK
** checking whether the namespace can be unloaded cleanly ... OK
** checking loading without being on the library search path ... OK
** checking use of S3 registration ... OK
* loading checks for arch 'x64'
** checking whether the package can be loaded ... OK
** checking whether the package can be loaded with stated dependencies ... OK
** checking whether the package can be unloaded cleanly ... OK
** checking whether the namespace can be loaded with stated dependencies ... OK
** checking whether the namespace can be unloaded cleanly ... OK
** checking loading without being on the library search path ... OK
** checking use of S3 registration ... OK
* checking dependencies in R code ... OK
* checking S3 generic/method consistency ... OK
* checking replacement functions ... OK
* checking foreign function calls ... OK
* checking R code for possible problems ... [12s] OK
* checking Rd files ... OK
* checking Rd metadata ... OK
* checking Rd line widths ... OK
* checking Rd cross-references ... OK
* checking for missing documentation entries ... OK
* checking for code/documentation mismatches ... OK
* checking Rd \usage sections ... OK
* checking Rd contents ... OK
* checking for unstated dependencies in examples ... OK
* checking installed files from 'inst/doc' ... OK
* checking files in 'vignettes' ... OK
* checking examples ...
** running examples for arch 'i386' ... [4s] OK
** running examples for arch 'x64' ... [5s] OK
* checking for unstated dependencies in 'tests' ... OK
* checking tests ...
** running tests for arch 'i386' ... [74s] ERROR
  Running 'testthat.R' [74s]
Running the tests in 'tests/testthat.R' failed.
Complete output:
  > library(testthat)
  > library(precommit)
  > test_check("precommit")
  == Skipped tests ===============================================================
  * On CRAN (2)
  * env not supported in system2 on Windows (2)
  * not_conda() is TRUE (1)
  
  == Failed tests ================================================================
  -- Failure (test-all.R:36:1): (code run outside of `test_that()`) --------------
  Expected: No error. Found:
  Error in style(...) : unused argument (base_indention = 0)
  Calls: tryCatch ... <Anonymous> -> transform_files -> make_transformer -> force
  Execution halted
  Backtrace:
      x
   1. \-precommit:::run_test(...) test-all.R:36:0
   2.   \-precommit:::run_test_impl(...)
  -- Failure (test-all.R:46:1): (code run outside of `test_that()`) --------------
  Expected: No error. Found:
  Error in style(...) : unused argument (base_indention = 0)
  Calls: tryCatch ... <Anonymous> -> transform_files -> make_transformer -> force
  Execution halted
  Backtrace:
      x
   1. \-precommit:::run_test(...) test-all.R:46:0
   2.   \-precommit:::run_test_impl(...)
  -- Failure (test-all.R:57:1): (code run outside of `test_that()`) --------------
  Expected: No error. Found:
  Error in style(...) : unused argument (base_indention = 4)
  Calls: tryCatch ... <Anonymous> -> transform_files -> make_transformer -> force
  Execution halted
  Backtrace:
      x
   1. \-precommit:::run_test(...) test-all.R:57:0
   2.   \-precommit:::run_test_impl(...)
  -- Failure (test-exec.R:21:3): Path can be derived for windows Python >= 3.0 ---
  path_derive_precommit_exec_win_python3plus_base() is empty.
  -- Failure (test-exec.R:22:3): Path can be derived for windows Python >= 3.0 ---
  fs::path_file(path_derive_precommit_exec_win()) not equal to precommit_executable_file().
  1/1 mismatches
  x[1]: ""
  y[1]: "pre-commit.exe"
  
  [ FAIL 5 | WARN 0 | SKIP 5 | PASS 82 ]
  Error: Test failures
  Execution halted
** running tests for arch 'x64' ... [80s] ERROR
  Running 'testthat.R' [80s]
Running the tests in 'tests/testthat.R' failed.
Complete output:
  > library(testthat)
  > library(precommit)
  > test_check("precommit")
  == Skipped tests ===============================================================
  * On CRAN (2)
  * env not supported in system2 on Windows (2)
  * not_conda() is TRUE (1)
  
  == Failed tests ================================================================
  -- Failure (test-all.R:36:1): (code run outside of `test_that()`) --------------
  Expected: No error. Found:
  Error in style(...) : unused argument (base_indention = 0)
  Calls: tryCatch ... <Anonymous> -> transform_files -> make_transformer -> force
  Execution halted
  Backtrace:
      x
   1. \-precommit:::run_test(...) test-all.R:36:0
   2.   \-precommit:::run_test_impl(...)
  -- Failure (test-all.R:46:1): (code run outside of `test_that()`) --------------
  Expected: No error. Found:
  Error in style(...) : unused argument (base_indention = 0)
  Calls: tryCatch ... <Anonymous> -> transform_files -> make_transformer -> force
  Execution halted
  Backtrace:
      x
   1. \-precommit:::run_test(...) test-all.R:46:0
   2.   \-precommit:::run_test_impl(...)
  -- Failure (test-all.R:57:1): (code run outside of `test_that()`) --------------
  Expected: No error. Found:
  Error in style(...) : unused argument (base_indention = 4)
  Calls: tryCatch ... <Anonymous> -> transform_files -> make_transformer -> force
  Execution halted
  Backtrace:
      x
   1. \-precommit:::run_test(...) test-all.R:57:0
   2.   \-precommit:::run_test_impl(...)
  -- Failure (test-exec.R:21:3): Path can be derived for windows Python >= 3.0 ---
  path_derive_precommit_exec_win_python3plus_base() is empty.
  -- Failure (test-exec.R:22:3): Path can be derived for windows Python >= 3.0 ---
  fs::path_file(path_derive_precommit_exec_win()) not equal to precommit_executable_file().
  1/1 mismatches
  x[1]: ""
  y[1]: "pre-commit.exe"
  
  [ FAIL 5 | WARN 0 | SKIP 5 | PASS 82 ]
  Error: Test failures
  Execution halted
* checking for unstated dependencies in vignettes ... OK
* checking package vignettes in 'inst/doc' ... OK
* checking re-building of vignette outputs ... [2s] OK
* checking PDF version of manual ... OK
* checking for detritus in the temp directory ... OK
* DONE
Status: 2 ERRORs, 1 NOTE
```

**Comment by jimhester** [comment]
I don't think this is an issue in rcmdcheck. `R CMD check` installs the package being checked in a temporary library and includes this library in the libpath envvars. However on windows this library envvar does not seem to be either inherited or used by processes forked from the test process. We have seen this behavior in a number of other places and the solution is as you mentioned to install the package before running the check.

It is possible that win builder has its libraries setup in such a way that the package being checked is being installed into the system library, which is why you don't see this issue there.

**Comment by lorenzwalthert** [comment]
Thanks Jim, I guess we can close this then.
